core/primitives/src/trie_key.rs

```diff
@@ -178,6 +180,9 @@ pub enum TrieKey {
         index: u64,
     },
     BandwidthSchedulerState,
+    OutgoingBufferReceiptSizes {
+        to_shard: ShardId,
+    },
```

Comment:
You keep the entire structure in a single (per shard) trie value. This means that every access will need to retrieve and parse the entire thing adding to the state witness size. How large is it? If it is non-trivial I would suggest implementing a more fine grained storage like the `indices` + `values` structure for storing receipts in state.

Reply:
nits: 
- The `BufferedReceipt` does not have the Outgoing prefix - I would apply the same convention here to make them more similar
- Maybe replace Sizes with groups to make it a bit more future proof? 
- How about `BufferedReceiptGroups`? Then we'll have `BufferedReceipt`, `BufferedReceiptIndices` and `BufferedReceiptGroups`.

Reply:
I guess we could have an optimized trie queue which keeps multiple groups in a single trie value, but that sounds even more complex.

Reply:
The framework for writing NEAR contracts has some neat wrappers for common datastructures that need to operate on top of trie, maybe we could add something similar in the internal runtime instead of implementing things manually each time.

Reply:
Another bad thing about the current struct is that it's not rollbackable, which is a bit unsettling. But I looked at the uses of `forward_or_buffer_receipt` and it didn't look like we rollback after sending out the receipt at any point. I guess congestion info is also not rollbackable.

Reply:
I guess if we read all of it every time it doesn't matter that much if it's a single or multiple nodes, if anything multiple makes it worse. Have you considered any other data structures for maintaining aggregate size info about receipts?

Reply:
> Have you considered any other data structures for maintaining aggregate size info about receipts?

I couldn't find anything better than this. The problem is that we want to have sizes for some prefixes from 0 to n, and we want these sizes to increase by smallish amounts (~100kB). But if you have any cool ideas I'd love to explore them.

Reply:
Well I guess you maybe could maintain something like an AVL tree, and walk down to subtrees until you hit some size, but that sounds super complex.

Reply:
I have a couple of ideas I'd like to discuss in the meeting